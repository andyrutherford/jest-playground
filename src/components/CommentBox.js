import React, { Component } from 'react';
import { connect } from 'react-redux';
import * as actions from 'actions';

class CommentBox extends Component {

  state = { comment: '' };

  // When component just gets rendered
  componentDidMount() {
    this.shouldNavigateAway();
  }

  // Any time component receives new props
  componentDidUpdate() {
    this.shouldNavigateAway();
  }

  shouldNavigateAway() {
    if (!this.props.auth) {
      this.props.history.push('/');
    }
  }

  handleChange = e => {
    this.setState({ comment: e.target.value });
  };

  handleSubmit = e => {
    e.preventDefault();
    this.props.saveComment(this.state.comment);
    this.setState({ comment: '' });
  };



  render() {
    return (
      <div>
        <form onSubmit = {this.handleSubmit}>
          <h4>Add a Comment</h4>
          <textarea 
            value = {this.state.comment}
            onChange = {this.handleChange} />
          <div>
            <button>Submit Comment</button>
          </div>
        </form>
        <button 
          className="fetch-comments"
          onClick={this.props.fetchComments}>Fetch Comments</button>
        <button
          className="clear-comments"
          onClick={this.props.clearComments}>Clear All</button>
      </div>

    );
  }
}

function mapStateToProps(state) {
  return { auth: state.auth };
}

export default connect(mapStateToProps, actions)(CommentBox);