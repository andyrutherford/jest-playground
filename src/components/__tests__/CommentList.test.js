import React from 'react';
import ReactDOM from 'react-dom';
import CommentBox from '../CommentBox';
import { shallow } from 'enzyme';

it('should render correctly', () => {
  const wrapper = shallow(<CommentBox />);
  expect(wrapper).toMatchSnapshot();
});
