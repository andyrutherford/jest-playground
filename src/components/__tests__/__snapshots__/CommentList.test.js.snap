// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CommentBox />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h4>
          Add a Comment
        </h4>,
        <textarea
          onChange={[Function]}
          value=""
        />,
        <div>
          <button>
            Submit Comment
          </button>
        </div>,
      ],
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Add a Comment",
        },
        "ref": null,
        "rendered": "Add a Comment",
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "onChange": [Function],
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "textarea",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <button>
            Submit Comment
          </button>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Submit Comment",
          },
          "ref": null,
          "rendered": "Submit Comment",
          "type": "button",
        },
        "type": "div",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h4>
            Add a Comment
          </h4>,
          <textarea
            onChange={[Function]}
            value=""
          />,
          <div>
            <button>
              Submit Comment
            </button>
          </div>,
        ],
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Add a Comment",
          },
          "ref": null,
          "rendered": "Add a Comment",
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "onChange": [Function],
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "textarea",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button>
              Submit Comment
            </button>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Submit Comment",
            },
            "ref": null,
            "rendered": "Submit Comment",
            "type": "button",
          },
          "type": "div",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
